File: 0853-car-fleet/0853-car-fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        '''
        given three parameters
        '''
        if len(position) == 1 and len(speed) == 1: return 1
        
        pair = [[p,s] for p,s in zip(position, speed)]
        
        stack = []
        
        for p,s in sorted(pair)[::-1]:
            stack.append((target-p)/s)
            if len(stack) >= 2 and stack[-1] <= stack[-2]:
                stack.pop()
        return len(stack)

File: 0853-car-fleet/0853-car-fleet.js
/**
 * @param {number} target
 * @param {number[]} position
 * @param {number[]} speed
 * @return {number}
 */
var carFleet = function(target, position, speed) {
    /*
    target = 12, position = [10,8,0,5,3], speed = [2,4,1,1,3]
    
    
    */
    
    let cars = [];
    for(let i=0; i<position.length; i++){
        cars.push({ 
            pos: position[i],
            speed: speed[i]
        })
    }
    
    cars.sort((a,b) => b.pos - a.pos);
    
    let stack = [];
    
    for(let i=0; i<cars.length; i++){
        const car = cars[i];
        const time = (target - car.pos) / car.speed;
        
        if(stack.length === 0 || time > stack[stack.length -1]){
            stack.push(time)
        }
    }
    return stack.length
};